src/openrct2-ui/windows/Ride.cpp

```diff
@@ -5974,6 +5979,9 @@ static void window_ride_graphs_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi
     // Plot
     int32_t x = dpi->x;
     int32_t top, bottom;
+    // Uses the limits (used to draw high forces in red on measurement tab) to determine if line should be drawn red.
+    // By default they are kept 0 unless the graph type supports it.
+    int32_t redThresholdPositive = 0, redThresholdNegative = 0;
```

Comment:
Please rename these to make it clearer they are related to the intensity stats. Something like `intensityThreasholdPositive`.
Also I think this is a good place to use `std::optional`, so that you don't have to rely on a zero-value.

Reply:
Not really sure how to use std::optional offhand to check if it's assigned or not, but I renamed the variable.

Reply:
Instead of using std::optional, the checks for a zero value could be replaced with `listType == GRAPH_VERTICAL || listType == GRAPH_LATERAL`, which I have done now in 2bcf08b.